src/pint/models/noise_model.py

```diff
@@ -474,20 +517,18 @@ def pl_rn_basis_weight_pair(self, toas):
         the dataset.
 
         """
-        tbl = toas.table
-        t = (tbl["tdbld"].quantity * u.day).to(u.s).value
-        amp, gam, nf = self.get_pl_vals()
-        Fmat, f = create_fourier_design_matrix(t, nf)
-        weight = powerlaw(f, amp, gam) * f[0]
-        return (Fmat, weight)
+        return (self.get_basis(toas), self.get_weights(toas))
 
     def pl_rn_cov_matrix(self, toas):
         Fmat, phi = self.pl_rn_basis_weight_pair(toas)
         return np.dot(Fmat * phi[None, :], Fmat.T)
 
 
-def create_quantization_matrix(toas_table, dt=1, nmin=2):
-    """Create quantization matrix mapping TOAs to observing epochs."""
+def get_ecorr_epochs(toas_table, dt=1, nmin=2):
+    """Find only epochs with more than 1 TOA for applying ECORR."""
```

Comment:
Does this really not return epochs with only one TOA? I would expect ECORR to be applied to those too.

Reply:
The existing implementation does the same thing. One of the tests I've included (test_ecorrnoise_basis_weights) checks that the new code produces results identical to the existing code. 

The relevant code is essentially identical to the code in ENTERPRISE. I think someone copy-pasted it.
https://github.com/nanograv/enterprise/blob/2fb1bf3cdc086ebe10da4689a7312fdda54e47ab/enterprise/signals/utils.py#L744

Reply:
There is a comment there that explicitly says "find only epochs with more than 1 TOA".